src/v1.rs

```diff
@@ -43,34 +44,69 @@ impl EszipV1 {
   pub fn get_module(&self, specifier: &str) -> Option<Module> {
     let mut specifier = &Url::parse(specifier).ok()?;
     let mut visited = HashSet::new();
+    let modules = self.modules.lock().unwrap();
     loop {
       visited.insert(specifier);
-      let module = self.modules.get(specifier)?;
+      let module = modules.get(specifier)?;
       match module {
         ModuleInfo::Redirect(redirect) => {
           specifier = redirect;
           if visited.contains(specifier) {
             return None;
           }
         }
-        ModuleInfo::Source(source) => {
+        ModuleInfo::Source(..) => {
           let module = Module {
             specifier: specifier.to_string(),
             kind: ModuleKind::JavaScript,
-            inner: ModuleInner::V1(Arc::new(
-              source
-                .transpiled
-                .as_ref()
-                .unwrap_or(&source.source)
-                .as_bytes()
-                .to_owned(),
-            )),
+            inner: ModuleInner::V1(EszipV1 {
+              version: self.version,
+              modules: self.modules.clone(),
+            }),
           };
           return Some(module);
         }
       }
     }
   }
+
+  pub(crate) fn get_module_source(
+    &self,
+    specifier: &str,
+  ) -> Option<Arc<Vec<u8>>> {
+    let specifier = &Url::parse(specifier).ok()?;
+    let modules = self.modules.lock().unwrap();
+    let module = modules.get(specifier).unwrap();
+    match module {
+      ModuleInfo::Redirect(_) => panic!("Redirects should be resolved"),
+      ModuleInfo::Source(module) => {
+        let source = module.transpiled.as_ref().unwrap_or(&module.source);
+        if source.is_empty() {
+          None
+        } else {
+          Some(Arc::new(source.clone().into_bytes()))
+        }
+      }
+    }
+  }
+
+  pub(crate) fn take_module_source(
+    &self,
+    specifier: &str,
+  ) -> Option<Arc<Vec<u8>>> {
+    let specifier = &Url::parse(specifier).ok()?;
+    let mut modules = self.modules.lock().unwrap();
+    let module = modules.get_mut(specifier).unwrap();
```

Comment:
I wonder whether this should directly be a `modules.remove(specifier)`?
It seems to leave the store in a weird state where the rest of the logic should now probe for `source.is_empty()`.
Or perhaps the new state should be encoded as a new `ModuleInfo::SourceTaken` variant?

Reply:
Yes, I switched it to modules.remove() since we don't need to preserve module metadata for v1.